lib/datasets_test.go

```diff
@@ -1116,6 +1116,50 @@ Pirates of the Caribbean: At World's End ,foo
 	}
 }
 
+func TestDatasetRequestsValidateFSI(t *testing.T) {
+	ctx, done := context.WithCancel(context.Background())
+	defer done()
+
+	mr, err := testrepo.NewTestRepo()
+	if err != nil {
+		t.Fatalf("error allocating test repo: %s", err.Error())
+	}
+	node, err := p2p.NewQriNode(mr, config.DefaultP2PForTesting(), event.NilBus, nil)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	inst := NewInstanceFromConfigAndNode(ctx, config.DefaultConfigForTesting(), node)
+
+	// we need some fsi stuff to fully test remove
+	methods := NewFSIMethods(inst)
+	// create datasets working directory
+	datasetsDir, err := ioutil.TempDir("", "QriTestValidateFSIDataset")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(datasetsDir)
```

Comment:
Check out TestRenameNoHistory in this file; you can replace all of this code above by creating a testRunner:

```
	tr := newTestRunner(t)
	defer tr.Delete()
	datasetsDir := tr.CreateAndChdirToWorkDir("QriTestValidateFSIDataset")
        methods := NewFSIMethods(tr.Instance)
```